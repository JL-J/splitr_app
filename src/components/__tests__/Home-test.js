// __tests__/Home.js
import 'react-native';
import React from 'react';
import Home from '../home.js';
import SubmitButton from '../submitButton.js'

import renderer from 'react-test-renderer';

describe('<Home />', () => {
  it('home renders correctly', () => {
    const snap = renderer.create(
      <Home />
      ).toJSON();
    expect(snap).toMatchSnapshot();
  });

  it('home renders a button', () => {
    const snap = renderer.create(
      <SubmitButton />
      ).toJSON();
    expect(snap).toMatchSnapshot();
  });

});
