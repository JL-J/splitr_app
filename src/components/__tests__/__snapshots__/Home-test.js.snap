// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> home renders a button 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#841584",
          },
        ]
      }
    >
      Submit
    </Text>
  </View>
</View>
`;

exports[`<Home /> home renders a text input box for names 1`] = `
<TextInput
  allowFontScaling={true}
  placeholder="Name"
  style={
    Object {
      "borderColor": "gray",
      "borderWidth": 1,
      "height": 40,
    }
  }
  underlineColorAndroid="transparent"
/>
`;

exports[`<Home /> home renders correctly 1`] = `
<View>
  <Text>
    Hello World!
  </Text>
  <TextInput
    allowFontScaling={true}
    placeholder="Name"
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
